### PR TITLE
Make the method test in class Engine return top1/top5/mAP for the main model

### PR DESCRIPTION
**Pull request by LeonidBeynenson**
Earlier it returned top1/top5/mAP for the last model, so
if mutual learning is used, it returned accuracy for the last auxiliary model.

### CODE REVIEW

**Comment by kprokofi** [comment]
@LeonidBeynenson , Have you already tested this? Could we mark this PR as ready for review?

**Comment by kprokofi** [comment]
Also, please, consider removing the test before running. It harms training time with LR finder. Each time before new considered learning rate this algorithm launches a test function.  Anyway, I don't see the need for testing before training. For what reason? If it would work when we launch a resume, I understand, but for now, it's pointless.

Then, every epoch it prints "Unfreezing aux models". We do not this print each time I guess.

**Comment by LeonidBeynenson** [comment]
>Also, please, consider removing the test before running. It harms training time with LR finder.
It is highly useful for NNCF trainings and for several other cases. Otherwise we do not know the initial quality when we start from pretrained weights. But I agree that it should not work during LR finder work.
How could we detect if the train function is run inside LR finder?
I think, this way is ok: `if not lr_finder: ...`

> Then, every epoch it prints "Unfreezing aux models". We do not this print each time I guess.
Ok!

**Comment by kprokofi** [comment]
> > Also, please, consider removing the test before running. It harms training time with LR finder.
> > It is highly useful for NNCF trainings and for several other cases. Otherwise we do not know the initial quality when we start from pretrained weights. But I agree that it should not work during LR finder work.
> > How could we detect if the train function is run inside LR finder?
> > I think, this way is ok: `if not lr_finder: ...`

>  Yes, that's okay, just if statement.

**Comment by LeonidBeynenson** [comment]
@kprokofi : Kirill, I think I have finished work on this PR
Could you merge it?

**Review by kprokofi — APPROVED** [review]
